src/main/webapp/app/exercises/programming/manage/grading/tasks/programming-exercise-task.ts

```diff
@@ -12,4 +12,6 @@ export class ProgrammingExerciseTask extends ProgrammingExerciseServerSideTask {
     resultingPoints?: number;
     resultingPointsPercent?: number;
     stats: TestCaseStats | undefined;
+
+    [key: string]: number | string | undefined | TestCaseStats | ProgrammingExerciseTestCase[];
```

Comment:
This also sticks out a bit. Why do we need to define for arbitrary strings that a programming exercise task might have attributes of type `number | string | undefined | TestCaseStats | ProgrammingExerciseTestCase[]`?

Reply:
@SimonEntholzer

Reply:
I remove the type and the client and e2e tests are still running through, also double checked the programming grading table and it worked, I am quite confident that the type was not necessary

Reply:
Might have been an IDE related issue on my side. It looked like without this additional attribute accessing some of the ProgrammingExerciseTask with the ....["..."] notation wouldn't work. Seems like it does work without the additional key attribute :+1: